src/iris/ui/static/js/iris.js

```diff
@@ -1762,6 +1777,27 @@ iris = {
       });
       this.redrawApplicationDropdown();
     },
+    createMobile: function() {
+      $.ajax({
+        url: this.data.mobileKeyUrl + this.data.user,
+        method: 'GET',
+        dataType: 'html'
+      }).done(function(data){
+        $('#mobile-key-container').html('<img width="300" height="300" src="' + data + '">')
```

Comment:
this is vulnerable to XSS in case `data` gets compromised. you should create the image object and programmatically assign the src value. eg:

```
$('#mobile-key-container').empty(); // clear it first if needed
$('#mobile-key-container').append($('<img width="300" height="300">').attr('src', data));
```

it's possible that @shoxv is aware of a less verbose way of doing this, but the gist is that html() is extremely dangerous, especially when interpolating data directly. It's pretty much HTML's equivalent to SQL injection